pkg/xds/generator/dns_generator.go

```diff
@@ -26,8 +26,16 @@ func (g DNSGenerator) Generate(ctx xds_context.Context, proxy *core_xds.Proxy) (
 	}
 	ipV6Enabled := proxy.Dataplane.Spec.GetNetworking().GetTransparentProxying().GetRedirectPortInboundV6() != 0
 
+	outboundIPs := map[string]bool{}
+	for _, outbound := range proxy.Dataplane.Spec.Networking.GetOutboundInterfaces() {
+		outboundIPs[outbound.DataplaneIP] = true
+	}
+
 	vips := map[string][]string{}
 	for _, dnsOutbound := range ctx.Mesh.VIPDomains {
+		if !outboundIPs[dnsOutbound.Address] {
+			continue // if there is no outbound for given address, there is no point of providing DNS resolver
+		}
 		for _, domain := range dnsOutbound.Domains {
 			v6 := util_net.ToV6(dnsOutbound.Address)
```

Comment:
Weird all this could be done outside the loop no?

Somehting like:
```go
addresses := []string{dnsOutbound.Address}
v6 := util_net.ToV6(dnsOutbound.Address)
if v6 != dnsOutbound.Address {
    addresses = append(addresses, dnsOutbound.Address)
}
if ipV6Enabled {
   addresses = append(addresses, v6)
}
for _, domain := range dnsOutbound.Domains {
   vips[domain] = addresses
}
```

Reply:
the code is slightly different but done